### PR TITLE
Update architecture

### PR DESCRIPTION
**Pull request by tk-o**
This PR simplifies architecture by relying more on composition. Thanks to that it was possible to remove top-level context container that was implicitly used everywhere (and hence it was hard to reason about its primary responsibility).

Furthermore, this PR integrates secret client to the new `/docs` page.

Example link:
- https://update-architecture.deploycontracts-io.pages.dev/docs?token=secret1708kv46qp929g0xd8rnhk2z7pxskn57qcp95kq

### CODE REVIEW

File: components/snip-20/SummaryCardWrapper.tsx [hunk 1]
```diff
@@ -21,9 +19,9 @@ export default function SummaryCardWrapper({ children, img, title, linkUrl }: Pr
           <div className='text-lg text-white font-space-grotesk font-bold'>{title}</div>
         </div>
 
-        <GradientText className='mt-5 sm:mt-0' onClick={() => router.push(linkUrl)}>
-          Edit
-        </GradientText>
+        <Link href={linkUrl}>
+          <GradientText className='mt-5 sm:mt-0'>Edit</GradientText>
+        </Link>
```
Comment (tk-o) [inline components/snip-20/SummaryCardWrapper.tsx:24]:
```suggestion
        <Link href={linkUrl}>
          <a><GradientText className='mt-5 sm:mt-0'>Edit</GradientText></a>
        </Link>
```

File: components/snip-20/tokenSummary.tsx [hunk 2]
```diff
@@ -18,36 +31,44 @@ export default function tokenSummary() {
...
-          <OutputDataRow title='Minter address' data='fq412t41g41b2b34n4mn24n323' />
-          <OutputDataRow title='Token name' data={snip20FormData[0].tokenName} />
-          <OutputDataRow title='Total supply' data={snip20FormData[0].tokenTotalSupply} />
+        <SummaryCardWrapper
+          linkUrl={stepPath(TokenCreatorStep.BasicInfo)}
+          img='/assets/step1-visited.svg'
+          title={stepTitles[0]}
+        >
+          <OutputDataRow title='Minter address' data={formData.basicTokenInfo.minterAddress} />
+          <OutputDataRow title='Token name' data={formData.basicTokenInfo.tokenSymbol} />
```
Comment (tk-o) [inline components/snip-20/tokenSummary.tsx:40]:
```suggestion
          <OutputDataRow title='Token symbol' data={formData.basicTokenInfo.tokenSymbol} />
```

File: consts.ts [hunk 3]
```diff
@@ -8,10 +10,24 @@ export const landingPageUrl = {
...
+  chainName: process.env.NEXT_PUBLIC_CHAIN_NAME!,
+  rpcUrl: process.env.NEXT_PUBLIC_CHAIN_RPC!,
+  grpcUrl: process.env.NEXT_PUBLIC_CHAIN_GRPC!,
+  restUrl: process.env.NEXT_PUBLIC_CHAIN_REST!,
+}
+
+export const CONTRACT_INFO = {
+  codeId: parseInt(process.env.NEXT_PUBLIC_CODE_ID!, 10),
+  codeHash: process.env.NEXT_PUBLIC_CODE_HASH!,
+}
```
Comment (tk-o) [inline consts.ts:31]:
```suggestion

```

**Review by deployAt — APPROVED** [review]
great job 🚀

**Pull request merged**